src/api/user/user.controller.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { UserController } from './user.controller';
+import { CreateUserDto } from './user.dto';
+import { User } from './user.entity';
+import { UserService } from './user.service';
+
+describe('UserController', () => {
+  let controller: UserController;
+  let service: UserService;
+
+  const USER_REPOSITORY_TOKEN = getRepositoryToken(User);
+
+  beforeEach(async () => {
+    const UserServiceProvider = {
+      provide: UserService,
+      useFactory: () => ({
+        findAll: jest.fn(),
+        getUser: jest.fn(),
+        createUser: jest.fn(() => []),
+      }),
+    };
+
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [UserController],
+      providers: [
+        UserService,
+        UserServiceProvider,
+        {
+          provide: USER_REPOSITORY_TOKEN,
+          useValue: {
+            create: jest.fn(),
+            save: jest.fn(),
+            find: jest.fn(),
+            findOneBy: jest.fn(),
+          },
+        },
+      ],
+    }).compile();
+
+    controller = module.get<UserController>(UserController);
+    service = module.get<UserService>(UserService);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+
+  describe('getAllUsers', () => {
+    it('calls service.findAll method', async () => {
+      controller.getAllUsers();
+      expect(service.findAll).toHaveBeenCalled();
+    });
+  });
+
+  describe('getUser', () => {
+    it('should return one users', async () => {
+      const result = { id: 1, username: 'Bill', isAdmin: true };
+      jest.spyOn(service, 'getUser').mockResolvedValue(result);
```

Comment:
This test could be improved by asserting that the service method is called with the expected arguments as well. I'm typically not a huge fan of asserting on the mocked value because that only tests that your mocks are wired up correctly. and result could be anything.